File: Python/Programs-based-on-Numbers/27-octal-to-decimal-conversion.py
octal = input("Enter Octal number: ")
decimal = 0
power = 0

for digit in octal[::-1]:
    decimal += int(digit) * (8**power)
    power += 1

print("Decimal value is:", decimal)


# Using built-in method
# decimal = int(octal, 8)

File: Java/Programs-based-on-Numbers/27-octal-to-decimal-conversion.java
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner getInput = new Scanner(System.in);

        System.out.println("Enter Octal number:");
        String octal = getInput.next();

        int decimal = 0;
        int power = 0;

        for (int i = octal.length()-1; i>=0; i--) {
            int digit = octal.charAt(i)-'0';
            decimal += digit * Math.pow(8, power);
            power++;
        }
        System.out.println("Decimal value is: " + decimal);
        getInput.close();
    }
}

// Using built-in method
// int decimal = Integer.parseInt(octal, 2);